include/test/mat_graph_verifier.h

```diff
@@ -28,4 +28,27 @@ class MatGraphVerifier : public GraphVerifier {
    * @return an array of connected components.
    */
   static std::vector<std::set<node_t>> kruskal(node_t n, const std::vector<bool>& input);
+
+  // Returns a unique identifier for each edge using diagonalization
+  static inline uint64_t get_uid(node_t i, node_t j) {
```

Comment:
is this any different from what we're doing in `util.cpp`?

Reply:
Yea, the one in util.cpp is now optimized for speed while this one is optimized for space

Reply:
(The one in util just concatenates the node_ids to get an edge_id)

Reply:
The one in `util.cpp` causes our data structure to be slightly larger then right? It's probably just `n(log^3(n^2) - log^3(n^2-n))` so not the end of the world.

Reply:
It does indeed but not when node ids are within 32bits. This might be something to revisit when we go up to 64 bit ids.